src/lava/lib/peripherals/realsense/realsense.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+
+import numpy as np
+import os
+import cv2
+
+import pyrealsense2 as rs
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.ports.ports import OutPort
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+
+
+class DirectRealsenseInput(AbstractProcess):
+    """
+    Outputting frames of RGB and Depth from Realsense camera
+    fetched from Realsense SDK or from recording files.
+    """
+
+    def __init__(
+        self,
+        height: int,
+        width: int,
+        filename: str = "",
+    ) -> None:
+        super().__init__(
+            height=height,
+            width=width,
+            filename=filename,
+        )
+        self.color_frame_out = OutPort(shape=(height, width, 3))
+        self.depth_frame_out = OutPort(shape=(height, width))
+
+
+@implements(proc=DirectRealsenseInput, protocol=LoihiProtocol)
+@requires(CPU)
+class DirectRealsenseInputPM(PyLoihiProcessModel):
+    color_frame_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, int)
+    depth_frame_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, int)
+
+    def __init__(self, proc_params):
+        super().__init__(proc_params)
+        self._height = proc_params["height"]
+        self._width = proc_params["width"]
+        self._filename = proc_params["filename"]
+        self._cur_steps = 0
+        if self._filename == "":
```

Comment:
What happens if a filename is provided? (what should happen?)

Reply:
Same as above